text/0000-tuple-struct-self-ctor.md

```diff
@@ -139,23 +224,116 @@ should work and will probably try at some point.
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-Rust (now) allows usage of `Self(v0, v1, ..)` inside inherent
-and trait `impl`s of tuple structs, either when mentioning the
-tuple struct directly in the `impl` header, or via a type alias.
+Rust (now) allows usage of `Self(v0, v1, ..)` inside inherent and
+trait `impl`s of tuple structs, either when mentioning the tuple struct
+directly in the `impl` header, or via a type alias.
+This usage is permitted both in expression and pattern contexts.
+
+Furthermore, `Self`, when referring to a tuple struct, can be legally coerced
```

Comment:
I think this basically follows from the description I gave before; it could be rewritten to say "This should allow `Self` to be legally coerced..."